app/src/main/java/com/cornellappdev/score/components/SportSelectorHeader.kt

```diff
@@ -0,0 +1,162 @@
+package com.cornellappdev.score.components
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.example.score.R
+import com.cornellappdev.android.score.data.models.Sport
+import com.cornellappdev.score.theme.CrimsonPrimary
+import com.cornellappdev.score.theme.GrayLight
+import com.cornellappdev.score.theme.GrayMid
+import com.cornellappdev.score.theme.Style.genderFilterText
+import com.cornellappdev.score.theme.Style.genderText
+import com.cornellappdev.score.theme.Style.sportFilterText
+import com.cornellappdev.score.theme.White
+
+@Preview
+@Composable
+fun PreviewSportSelectorHeader() {
+    val sports = listOf(
+        Triple("All", painterResource(id = R.drawable.ic_all), painterResource(id = R.drawable.ic_all_filled)),
+        Triple("Baseball", painterResource(id = R.drawable.ic_baseball), painterResource(id = R.drawable.ic_baseball_filled)),
+        Triple("Basketball", painterResource(id = R.drawable.ic_basketball), painterResource(id = R.drawable.ic_basketball_filled)),
+        Triple("Cross Country", painterResource(id = R.drawable.ic_cross_country), painterResource(id = R.drawable.ic_cross_country_filled)),
+        Triple("Football", painterResource(id = R.drawable.ic_football), painterResource(id = R.drawable.ic_football_filled))
+    )
+
+    SportSelectorHeader(sports, {}, {})
+}
+
+@Composable
+fun SportSelectorHeader(sports: List<Triple<String, Painter, Painter>>,
+                        onGenderSelected: (String) -> Unit,
+                        onSportSelected: (String) -> Unit) {
+    var selectedOption by remember { mutableStateOf("All") }
+    var selectedSport by remember { mutableStateOf(sports.first().first) }
```

Comment:
I'm a bit against having these as mutable states, since if we scroll away too far from the header for example, the state will be lost.

Let's take in the selectedOption/sport as an argument to the component, and expose a `onOptionSelected` and `onSportSelected`. This way, the viewmodel can handle the logic, which is more robust.